spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/ExpireSnapshotsSparkAction.java

```diff
@@ -184,11 +186,16 @@ public Dataset<FileInfo> expireFiles() {
 
       expireSnapshots.cleanExpiredFiles(false).commit();
 
-      // fetch metadata after expiration
-      Dataset<FileInfo> validFileDS = validFileDS(ops.refresh());
+      // fetch valid files after expiration
+      TableMetadata updatedMetadata = ops.refresh();
+      Dataset<FileInfo> validFileDS = fileDS(updatedMetadata);
+
+      // fetch files referenced by expired snapshots
+      Set<Long> removedSnapshotIds = findExpiredSnapshotIds(originalMetadata, updatedMetadata);
```

Comment:
Just noticed this variable name doesnt match spark 3.2, will change it.  (Initially it was 'expiredSnapshotIds' but it hides the member variable name and so I made last minute change).